src/aura/AuraAvatarUtils.sol

```diff
@@ -0,0 +1,163 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import {PRECISION, PRECISION_DOUBLE} from "../BaseConstants.sol";
+import {LogExpMath} from "../lib/balancer/LogExpMath.sol";
+import {AuraConstants} from "./AuraConstants.sol";
+
+import {IAuraToken} from "../interfaces/aura/IAuraToken.sol";
+import {IAsset} from "../interfaces/balancer/IAsset.sol";
+import {IBalancerVault} from "../interfaces/balancer/IBalancerVault.sol";
+import {IWeightedPool} from "../interfaces/balancer/IWeightedPool.sol";
+import {IPriceOracle} from "../interfaces/balancer/IPriceOracle.sol";
+import {IAggregatorV3} from "../interfaces/chainlink/IAggregatorV3.sol";
+
+contract AuraAvatarUtils is AuraConstants {
+    ////////////////////////////////////////////////////////////////////////////
+    // ERRORS
+    ////////////////////////////////////////////////////////////////////////////
+
+    error StalePriceFeed(uint256 currentTime, uint256 updateTime, uint256 maxPeriod);
+
+    ////////////////////////////////////////////////////////////////////////////
+    // INTERNAL VIEW
+    ////////////////////////////////////////////////////////////////////////////
+
+    function getAuraPriceInUsdSpot(uint256 _auraAmount) internal returns (uint256 usdPrice_) {
+        usdPrice_ = querySwapAuraForUsdc(_auraAmount) * AURA_USD_SPOT_FACTOR / _auraAmount;
+    }
+
+    function getBalAmountInUsdc(uint256 _balAmount) internal view returns (uint256 usdcAmount_) {
+        uint256 balInUsd = fetchPriceFromClFeed(BAL_USD_FEED, CL_FEED_HEARTBEAT_BAL);
+        // Divisor is 10^20 and uint256 max ~ 10^77 so this shouldn't overflow for normal amounts
+        usdcAmount_ = (_balAmount * balInUsd) / BAL_USD_FEED_DIVISOR;
+    }
+
+    function getAuraAmountInUsdc(uint256 _auraAmount, uint256 _twapPeriod)
+        internal
+        view
+        returns (uint256 usdcAmount_)
+    {
+        uint256 auraInEth = fetchPriceFromBalancerTwap(BPT_80AURA_20WETH, _twapPeriod);
+        uint256 ethInUsd = fetchPriceFromClFeed(ETH_USD_FEED, CL_FEED_HEARTBEAT_ETH_USD);
+        // Divisor is 10^38 and uint256 max ~ 10^77 so this shouldn't overflow for normal amounts
+        usdcAmount_ = (_auraAmount * auraInEth * ethInUsd) / AURA_USD_TWAP_DIVISOR;
+    }
+
+    function getBalAmountInBpt(uint256 _balAmount) internal view returns (uint256 bptAmount_) {
+        uint256 bptPriceInBal = getBptPriceInBal();
+        bptAmount_ = (_balAmount * PRECISION) / bptPriceInBal;
```

Comment:
A little nit but I think the variable name is a bit misleading, maybe it should be: `balAmountInBpt_`